src/test/java/community/solace/spring/cloud/requestreply/service/RequestReplyLoggingServiceTests.java

```diff
@@ -0,0 +1,436 @@
+package community.solace.spring.cloud.requestreply.service;
+
+import community.solace.spring.cloud.requestreply.AbstractRequestReplyLoggingIT;
+import community.solace.spring.cloud.requestreply.model.SensorReading;
+import community.solace.spring.cloud.requestreply.service.header.parser.errormessage.RemoteErrorException;
+import community.solace.spring.cloud.requestreply.service.logging.RequestReplyLogger;
+import org.junit.jupiter.api.Test;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.Mockito;
+import org.slf4j.LoggerFactory;
+import org.slf4j.event.Level;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.cloud.stream.binder.BinderHeaders;
+import org.springframework.cloud.stream.function.StreamBridge;
+import org.springframework.messaging.Message;
+import org.springframework.messaging.support.MessageBuilder;
+import org.springframework.messaging.support.MessageHeaderAccessor;
+import reactor.core.publisher.Flux;
+import reactor.test.StepVerifier;
+
+import java.time.Duration;
+import java.util.Map;
+import java.util.concurrent.TimeoutException;
+import java.util.concurrent.atomic.AtomicReference;
+
+import static community.solace.spring.cloud.requestreply.model.SensorReading.BaseUnit.CELSIUS;
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.ArgumentMatchers.*;
+
+class RequestReplyLoggingServiceTests extends AbstractRequestReplyLoggingIT {
+
+    @Captor
+    ArgumentCaptor<Message<?>> messageCaptor;
+    @Captor
+    ArgumentCaptor<String> destinationCaptor;
+    @MockBean
+    private StreamBridge streamBridge;
+    @Autowired
+    private RequestReplyServiceImpl requestReplyService;
+    @MockBean
+    private RequestReplyLogger requestReplyLogger;
+
+    @Test
+    void requestAndAwaitReplyToTopic_expectExceptionAndErrorLogToBeanLogger_whenNoResponse() {
+        SensorReading request = new SensorReading();
+        request.setSensorID("toilet");
+
+        assertThrows(TimeoutException.class, () -> requestReplyService.requestAndAwaitReplyToTopic(
+                request,
+                "last_value/temperature/celsius/demo",
+                SensorReading.class,
+                Duration.ofMillis(100)
+        ));
+
+        // at least one error log must be thrown into the customized logger:
+        Mockito.verify(requestReplyLogger, Mockito.times(1)).log(eq(LoggerFactory.getLogger(RequestReplyServiceImpl.class)),
+                eq(Level.ERROR),
+                anyString(),
```

Comment:
Improve: test for expected error. May be use captor and hamcrast string contains

Reply:
OR:
Mockito.matches("(.*apple.*banana.*)|(.*banana.*apple.*)"